desc/compute/_neoclassical.py

```diff
@@ -0,0 +1,626 @@
+        )
+
+        def fun(pitch_inv):
+            points = bounce.points(pitch_inv, num_well=num_well)
+            v_tau, f1, f2 = bounce.integrate(
+                [_v_tau, _f1, _f2],
+                pitch_inv,
+                data,
+                ["|grad(psi)|*kappa_g", "|B|_r|v,p"],
+                points,
+                is_fourier=True,
+            )
+            gamma_c = jnp.arctan(
+                safediv(
+                    f1,
+                    (
+                        f2
+                        + bounce.integrate(
+                            _f3,
+                            pitch_inv,
+                            data,
+                            "K",
+                            points,
+                            quad=quad2,
+                            is_fourier=True,
+                        )
+                    )
+                    * bounce.interp_to_argmin(
+                        data["|grad(rho)|*|e_alpha|r,p|"], points, is_fourier=True
+                    ),
+                )
+            )
+            return jnp.sum(v_tau * gamma_c**2, axis=-1)
+
+        return jnp.sum(
+            _foreach_pitch(fun, data["pitch_inv"], batch_size)
+            * data["pitch_inv weight"]
+            / data["pitch_inv"] ** 2,
+            axis=-1,
+        ) / bounce.compute_fieldline_length(fieldline_quad)
+
+    grid = transforms["grid"]
+    # It is assumed the grid is sufficiently dense to reconstruct |B|,
+    # so anything smoother than |B| may be captured accurately as a single
+    # Fourier series rather than transforming each component.
+    # Last term in K behaves as ∂log(|B|²/B^ϕ)/∂ρ |B| if one ignores the issue
+    # of a log argument with units. Smoothness determined by positive lower bound
+    # of log argument, and hence behaves as ∂log(|B|)/∂ρ |B| = ∂|B|/∂ρ.
+    data["Gamma_c"] = _compute(
+        Gamma_c,
+        fun_data={
+            "|grad(psi)|*kappa_g": data["|grad(psi)|"] * data["kappa_g"],
+            "|grad(rho)|*|e_alpha|r,p|": data["|grad(rho)|"] * data["|e_alpha|r,p|"],
+            "|B|_r|v,p": data["|B|_r|v,p"],
+            "K": data["iota_r"]
+            * dot(cross(data["grad(psi)"], data["b"]), data["grad(phi)"])
+            - (
+                2 * data["|B|_r|v,p"]
+                - data["|B|"] * data["B^phi_r|v,p"] / data["B^phi"]
+            ),
+        },
+        data=data,
+        theta=theta,
+        grid=grid,
+        num_pitch=num_pitch,
+        simp=False,
+    ) / (2**1.5 * jnp.pi)
+    return data
+
+
+def _cvdrift0(data, B, pitch):
+    return safediv(
+        data["cvdrift0"] * (1 - 0.5 * pitch * B),
+        jnp.sqrt(jnp.abs(1 - pitch * B)),
+    )
+
+
+def _gbdrift(data, B, pitch):
+    return safediv(
+        (data["gbdrift (periodic)"] + data["gbdrift (secular)/phi"] * data["zeta"])
+        * (1 - 0.5 * pitch * B),
+        jnp.sqrt(jnp.abs(1 - pitch * B)),
+    )
+
+
+@register_compute_fun(
+    name="Gamma_c Velasco",
```

Comment:
did we sort out whether this goes to zero for an infinite fieldline?

Reply:
There is no change in the status on this.

The secular term drives  `Gamma_c Velasco` to zero for an infinite fieldline. Dr. Velasco informed me they had proven the secular term is supposed to average out of the final result on paper. This may provide more motivation for #1428 . That being said the results don't change even at high resolution for fixed quadrature, and we know the quadratures are supposed to converge on paper and empirically.

Once #1406 udpates the equilibrium from #1369 I think it's worth checking again.

Reply:
Indeed, if I remove the secular term from the bounce integrals, the result converges to a nonzero profile quickly.